tools/run_tests/performance/bq_upload_result.py

```diff
@@ -157,6 +169,40 @@ def _populate_metadata_inplace(scenario_result):
     scenario_result['metadata'] = metadata
 
 
+def _populate_metadata_from_file(scenario_result, test_metadata_file):
+    with open(test_metadata_file, 'r') as f:
+        test_metadata = json.loads(f.read())
+
+    """eliminate managedField from metadata set"""
+    if 'managedFields' in test_metadata:
+        del test_metadata['managedFields']
+    annotations = test_metadata['annotations']
+
+    """if use kubectl apply ..., kubectl will append current configuration to
+     annotation, the field is deleted since it includes a lot of irrelevant 
+     information"""
+    if 'kubectl.kubernetes.io/last-applied-configuration' in annotations:
+        del annotations['kubectl.kubernetes.io/last-applied-configuration']
+
+    utc_timestamp = str(calendar.timegm(time.gmtime()))
+    metadata = {'created': utc_timestamp}
+
+    if 'buildNumber' in metadata:
```

Comment:
Here I would define (possibly somewhere else) a dictionary to do these transformations.

You could use some logic if you don't want to hardcode it:

```python
_annotation_to_metadata_key_map =_{
  'ci_' + key: key
  for key in (
    'buildNumber',
    'buildUrl',
    'jobName',
    'gitCommit' ,
    'gitActualCommit',
  )}
```

You would then have something like

```python
for  key, value in _annotation_to_bq_metadata_key_map.items():
  if key in annotations:
    metadata[value] = annotations[key]
```